deployment/jobservice/templates/statefulset.yaml

```diff
@@ -0,0 +1,72 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "jobservice.name" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    {{- include "jobservice.labels.all" . | nindent 4 }}
+spec:
+  replicas: {{ .Values.replicas }}
+  serviceName: {{ include "jobservice.name" . }}
+  selector:
+    matchLabels:
+      {{- include "jobservice.labels.identity" . | nindent 6 }}
+  template:
+    metadata:
+      name: {{ include "jobservice.name" . }}
+      labels:
+        {{- include "jobservice.labels.all" . | nindent 8 }}
+    spec:
+      terminationGracePeriodSeconds: {{ .Values.terminationGracePeriodSeconds }}
+      serviceAccountName: {{ .Values.customServiceAccount | default (include "jobservice.name" .) }}
+      securityContext:
+        runAsUser: 1000
+        runAsGroup: 2000
+      containers:
+        - name: jobservice
+          imagePullPolicy: IfNotPresent
+          image: {{ .Values.image.repository }}:{{ required "A value is required for .Values.image.tag" .Values.image.tag }}
+          args:
+            - run
+            - --config
+            - /config/application_config.yaml
+          {{- if .Values.env }}
+          env:
+            {{- toYaml .Values.env | nindent 12 -}}
+          {{- end }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          ports:
+            - containerPort:  {{ .Values.applicationConfig.grpcPort }}
+              protocol: TCP
+              name: grpc
+          volumeMounts:
+            - name: user-config
+              mountPath: /config/application_config.yaml
+              subPath: {{ include "jobservice.config.filename" . }}
+              readOnly: true
+            {{- if .Values.additionalVolumeMounts }}
+            {{- toYaml .Values.additionalVolumeMounts | nindent 12 -}}
+            {{- end }}
+
+          securityContext:
+            allowPrivilegeEscalation: false
+      affinity:
+        podAntiAffinity:
+          preferredDuringSchedulingIgnoredDuringExecution:
+            - weight: 100
+              podAffinityTerm:
+                labelSelector:
+                  matchExpressions:
+                    - key: app
+                      operator: In
+                      values:
+                        - {{ include "jobservice.name" . }}
+                topologyKey: kubernetes.io/hostname
+      volumes:
+        - name: user-config
+          secret:
+            secretName: {{ include "jobservice.config.name" . }}
+        {{- if .Values.additionalVolumes }}
```

Comment:
It'd be neat if we could support some standard pvc/emptydir setup. However maybe this is the standard approach for statefulsets publically.

Just a thought and nothing major

Reply:
We discussed and we will hold on this.  I'm unclear of how the PVC stuff will go with internal k8 so I'd like to hold on it.